java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java

```diff
@@ -302,6 +302,14 @@ public void clear() {
     valueCount = 0;
   }
 
+  @Override
+  public void reset() {
+    for (final ValueVector v : getChildren()) {
+      v.reset();
+    }
+    valueCount = 0;
```

Comment:
should this be changed to `setValueCount(0)` to also set count on children?

Reply:
That's not needed since the keeper of valueCount is not the child vector(s) but the top level container map vector. getValueCount() doesn't consult the child vectors for the valueCount and returns the valueCount tracked locally.